AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreEndToEndTests.swift

```diff
@@ -330,6 +330,79 @@ class DataStoreEndToEndTests: SyncEngineIntegrationTestBase {
         try validateSavePost()
     }
 
+    /// Perform concurrent saves and observe the data successfuly synced from cloud. Then delete the items afterwards
+    /// and ensure they have successfully synced from cloud
+    ///
+    /// - Given: DataStore is in ready state
+    /// - When:
+    ///    - Concurrently perform Save's
+    /// - Then:
+    ///    - Ensure the expected mutation event with version 1 (synced from cloud) is received
+    ///    - Clean up: Concurrently perform Delete's
+    ///    - Ensure the expected mutation event with version 2 (synced from cloud) is received
+    ///
+    func testConcurrentSave() throws {
+
+        try startAmplifyAndWaitForReady()
+
+        var posts = [Post]()
+        let count = 5
+        for _ in 0 ..< count {
+            let post = Post(title: "title",
+                            content: "content",
+                            createdAt: .now())
+            posts.append(post)
+        }
+        let postsSyncedToCloud = expectation(description: "All posts saved and synced to cloud")
+        var postsSyncedToCloudCount = 0
+        let postsDeletedFromCloud = expectation(description: "All posts deleted and synced to cloud")
+        var postsDeletedFromCloudCount = 0
+        let sink = Amplify.DataStore.publisher(for: Post.self).sink { completed in
+            switch completed {
+            case .finished:
+                break
+            case .failure(let error):
+                XCTFail("\(error)")
+            }
+        } receiveValue: { mutationEvent in
+            do {
+                let receivedPost = try mutationEvent.decodeModel(as: Post.self)
+                if mutationEvent.mutationType == MutationEvent.MutationType.create.rawValue,
+                   posts.contains(where: { $0.id == receivedPost.id }),
```

Comment:
should it be `$0.modelId` ?

Reply:
`posts` is data created above so there is no `modelId` here. i suppose we could also do
```swift
posts.contains(where: { $0.id == mutationEvent.modelid }
```

Reply:
yes, i think if `decodeModel` is expensive, we can try that.